libraries/AC_AutoTune/AC_AutoTune_Heli.cpp

```diff
@@ -698,205 +736,35 @@ void AC_AutoTune_Heli::report_axis_gains(const char* axis_string, float rate_P,
-            attitude_control->rate_bf_yaw_target(rate_request_cds.get());
-        } else if (((wrap_180_cd(ahrs_view->yaw_sensor - trim_heading) < -2.0f * max_angle_cd && is_positive(dir_sign))
-                   || (wrap_180_cd(ahrs_view->yaw_sensor - trim_heading) > 2.0f * max_angle_cd && !is_positive(dir_sign)))
-                   && ff_test_phase == 1 ) {
-            ff_test_phase = 2;
-            attitude_control->reset_yaw_target_and_rate(false);
-            angle_request_cd.reset(wrap_180_cd(ahrs_view->yaw_sensor - trim_heading));
-            attitude_control->input_euler_angle_roll_pitch_yaw(start_angles.x, start_angles.y, angle_request_cd.get(), false);
-        } else if (ff_test_phase == 2 ) {
-            angle_request_cd.apply(0.0f, AP::scheduler().get_loop_period_s());
-            attitude_control->input_euler_angle_roll_pitch_yaw(start_angles.x, start_angles.y, wrap_360_cd(trim_heading + angle_request_cd.get()), false);
-        }
-        break;
-    }
-
-    rotation_rate = rotation_rate_filt.apply(gyro_reading,
-                    AP::scheduler().get_loop_period_s());
-    command_out = command_filt.apply((command_reading - trim_command_reading),
-                                     AP::scheduler().get_loop_period_s());
-    filt_target_rate = target_rate_filt.apply(tgt_rate_reading,
-                       AP::scheduler().get_loop_period_s());
-
-    // record steady state rate and motor command
-    switch (axis) {
-    case ROLL:
-        if (((ahrs_view->roll_sensor >= -max_angle_cd + start_angle && is_positive(dir_sign))
-            || (ahrs_view->roll_sensor <= max_angle_cd + start_angle && !is_positive(dir_sign)))
-            && ff_test_phase == 1 ) {
-            test_rate_filt = rotation_rate;
-            test_command_filt = command_out;
-            test_tgt_rate_filt = filt_target_rate;
-        }
-        break;
-    case PITCH:
-        if (((ahrs_view->pitch_sensor >= -max_angle_cd + start_angle && is_positive(dir_sign))
-            || (ahrs_view->pitch_sensor <= max_angle_cd + start_angle && !is_positive(dir_sign)))
-            && ff_test_phase == 1 ) {
-            test_rate_filt = rotation_rate;
-            test_command_filt = command_out;
-            test_tgt_rate_filt = filt_target_rate;
-        }
-        break;
-    case YAW:
-    case YAW_D:
-        if (((wrap_180_cd(ahrs_view->yaw_sensor - trim_heading) >= -2.0f * max_angle_cd && is_positive(dir_sign))
-            || (wrap_180_cd(ahrs_view->yaw_sensor - trim_heading) <= 2.0f * max_angle_cd && !is_positive(dir_sign)))
-            && ff_test_phase == 1 ) {
-            test_rate_filt = rotation_rate;
-            test_command_filt = command_out;
-            test_tgt_rate_filt = filt_target_rate;
+    test_input_type = waveform_input_type;
+    test_freq_resp_input = freq_resp_input;
+    test_calc_type = calc_type;
+    test_start_freq = start_frq;
+    //target attitude magnitude
+    tgt_attitude = amplitude * 0.01745f;
```

Comment:
0.01745? pi / 180? If its not that a comment explaining what it is would be good.
```suggestion
    tgt_attitude = radians(amplitude);
```

Reply:
done. thanks.